tests/concepts/test_View.cpp

```diff
@@ -27,4 +32,56 @@ TEST(concepts, View)
     static_assert(Kokkos::utils::concepts::View<view_2d_t>);
 }
 
+//! @test Check @ref Kokkos::utils::concepts::ViewOfInstanceOf.
+TEST(KokkosTraits, is_view_of_instance_of)
+{
+    using Kokkos::utils::concepts::ViewOfInstanceOf;
+
+    static_assert(   ViewOfInstanceOf<Kokkos::View<std::vector<int> , execution_space>, std::vector>);
+    static_assert(   ViewOfInstanceOf<Kokkos::View<std::vector<int>*, execution_space>, std::vector>);
+    static_assert( ! ViewOfInstanceOf<Kokkos::View<std::vector<int>*, execution_space>, std::tuple>);
```

Comment:
May be good to avoid try to avoid any issues that could come from putting `std::vector` in a `View` that will be on device in cuda/hip cases. We could do e.g.

```
static_assert(   ViewOfInstanceOf<Kokkos::View<std::array<int, 2> , execution_space>, std::array>);
```

Reply:
OK, it's now using `std::tuple` and `Kokkos::IndexType`.